### ISSUE TITLE
Gather ideas and suggestions

### ISSUE THREAD

**Issue by s-tikhomirov** [s-tikhomirov/getroutevia#1]
TLDR: this plugin would implement `getroute` with additional requirements on nodes.

This idea originated while doing research on probing LN channels. I already have some initial implementation as part of the code accompanying [this recent paper](https://arxiv.org/abs/2004.00333) (the code is not published). (By the way, I'm going to present this research in a separate session on 2020-04-04 16:00 GMT+2 - [Jitsi link](https://meet.fulmo.org/LightningProbingSergeiTikhomirov)). The goal of this project is to separate the functionality related to choosing routes and release it a separate plugin which may be of value for goals unrelated to channel probing ;)

The way I envision this plugin is as follows. It would enable an additional command `getroute-via`, in addition to `getroute`. It will take as parameters:

* a list of nodes the route must go through
* a list of nodes the route must not go through
* the order of known nodes in some form (TODO: determine the format)
* some properties of nodes that all nodes must satisfy (IP only from / only not from a certain country based on IP?)
* the maximum number of iterations (see below)

Internally, the plugin would call `getroute` in a loop. If the first generated route does not satisfy the requirements, the "bad" channel is added to the list of "bad" channels, and `getroute` is called again with "bad" channels `exclude`d. The process stops if a "good" route is found or if the maximum number of iterations has been done.

Of course, this is a rather dumb approach. A better way to implement this would be to check the restrictions inside `getroute` (for example, remove the "bad" nodes / channels from the graph before running the path finding algorithm). However, I find the idea feasible as a humble hackathon project, with the primary goal being improving my understanding of the c-lightning plugin system :)

If you are interested in collaborating on this project, or have any ideas or suggestions, please comment here.
